common/minimodel/__init__.py

```diff
@@ -0,0 +1,474 @@
+
+class MagnitudeSystem(Enum):
+    VEGA = auto()
+    AB = auto()
+    JY = auto()
+
+
+@dataclass
+class MagnitudeBand:
+    """
+    Values for center and width should be specified in nanometers.
+    """
+    name: str
+    center: float
+    width: float
+    system: MagnitudeSystem = MagnitudeSystem.VEGA
+    description: Optional[str] = None
+
+
+class MagnitudeBands(MagnitudeBand, Enum):
+    """
+    It is unconventional to use lowercase characters in an enum, but to differentiate
+    them from the uppercase magnitude bands, we must.
+    """
+    u = MagnitudeBand('u', 356, 46, MagnitudeSystem.AB, 'UV')
+    g = MagnitudeBand('g', 483, 99, MagnitudeSystem.AB, 'green')
+    r = MagnitudeBand('r', 626, 96, MagnitudeSystem.AB, 'red')
+    i = MagnitudeBand('i', 767, 106, MagnitudeSystem.AB, 'far red')
+    z = MagnitudeBand('z', 910, 125, MagnitudeSystem.AB, 'near-infrared')
+    U = MagnitudeBand('U', 360, 75, description='ultraviolet')
+    B = MagnitudeBand('B', 440, 90, description='blue')
+    V = MagnitudeBand('V', 550, 85, description='visual')
+    UC = MagnitudeBand('UC', 610, 63, description='UCAC')
+    R = MagnitudeBand('R', 670, 100, description='red')
+    I = MagnitudeBand('I', 870, 100, description='infrared')
+    Y = MagnitudeBand('Y', 1020, 120)
+    J = MagnitudeBand('J', 1250, 240)
+    H = MagnitudeBand('H', 1650, 300)
+    K = MagnitudeBand('K', 2200, 410)
+    L = MagnitudeBand('L', 3760, 700)
+    M = MagnitudeBand('M', 4770, 240)
+    N = MagnitudeBand('N', 10470, 5230)
+    Q = MagnitudeBand('Q', 20130, 1650)
+    AP = MagnitudeBand('AP', 550, 85, MagnitudeSystem.VEGA, 'apparent')
+
+
+@dataclass
+class Magnitude:
+    band: MagnitudeBands
+    value: float
+    error: float
+
+
+class TargetType(Enum):
+    BASE = auto()
+    USER = auto()
+    BLIND_OFFSET = auto()
+    OFF_AXIS = auto()
+    TUNING_STAR = auto()
+    GUIDING = auto()
+    OTHER = auto()
+
+
+class GuideSpeed(IntEnum):
+    SLOW = auto()
+    MEDIUM = auto()
+    FAST = auto()
+
+
+class TargetTag(Enum):
+    COMET = auto()
+    ASTEROID = auto()
+    MAJOR_BODY = auto()
+
+
+@dataclass
+class Target(ABC):
+    """
+    RA and Dec should be specified in decimal degrees.
+    """
+    name: str
+    ra: float
+    dec: float
+    magnitudes: Set[Magnitude]
+    type: TargetType
+
+    def __post_init__(self):
+        pass
+
+    def guide_speed(self) -> GuideSpeed:
+        """
+        Calculate the guide speed for this target.
+        """
+        pass
+
+
+@dataclass
+class SiderealTarget(Target):
+    """
+    Proper motion must be specified in milliarcseconds / year.
+    Epoch must be the decimal year.
+    """
+    pm_ra: float
+    pm_dec: float
+    epoch: float
+
+    def __post_init__(self):
+        super().__post_init__()
+
+
+@dataclass
+class NonsiderealTarget(Target):
+    des: str
+    tag: TargetTag
+
+    def __post_init__(self):
+        super().__post_init__()
+
+
+@dataclass
+class Resource:
+    id: str
+    name: str
```

Comment:
If I recall the Selector would need more information (FPU, Gratings, etc). Is this going to be separate from the mini model ?